substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/TypeEntry.java

```diff
@@ -38,12 +34,24 @@
 import static com.oracle.objectfile.debuginfo.DebugInfoProvider.DebugTypeInfo.DebugTypeKind.INTERFACE;
 import static com.oracle.objectfile.debuginfo.DebugInfoProvider.DebugTypeInfo.DebugTypeKind.PRIMITIVE;
 
+import com.oracle.objectfile.debuginfo.DebugInfoProvider.DebugTypeInfo;
+import com.oracle.objectfile.debuginfo.DebugInfoProvider.DebugTypeInfo.DebugTypeKind;
+import com.oracle.objectfile.elf.dwarf.DwarfDebugInfo;
+
+import jdk.graal.compiler.debug.DebugContext;
+
 public abstract class TypeEntry {
     /**
      * The name of this type.
      */
     protected final String typeName;
 
+    /**
+     * The type signature of this type.
+     */
+    protected long typeSignature;
+    protected long indirectTypeSignature;
```

Comment:
A better name might be `typeSignatureForCompressed`

Reply:
Ideally we would not need a separate field at all for the compressed variants and be able to derive a compressed variant typeSignature from the one and only `typeSignature`. But we can fix this in a followup PR.